api/ttn/lorawan/v3/end_device.proto

```diff
@@ -493,6 +493,56 @@ message ADRSettings {
     }
 
     ChannelSteeringSettings channel_steering = 8;
+
+    // EXPERIMENTAL: Data rate index override settings.
+    message PerDataRateIndexOverride {
+      option (thethings.flags.message) = {
+        select: true,
+        set: true
+      };
+      reserved 1;
```

Comment:
I didn't really find a good reason to make the transmission power configurable on a per data rate index basis, and doing so would've required a larger rework of the ADR algorithm, so I left just left the fields as reserved for now.